content/docs/components/modal/theming.mdx

```diff
@@ -2,3 +2,198 @@
 id: modal
 scope: theming
 ---
+
+## Overview
+
+The `Modal` component is a multipart component.
+
+> To learn more about styling multipart components, visit the
+> [Component Style](/docs/styled-system/component-style#styling-multipart-components)
+> page.
+
+## Anatomy
+
+- `header`
+- `overlay`
+- `dialogContainer`
+- `dialog`
+- `closeButton`
+- `body`
+- `footer`
+
+You can find more information in the source
+[here](https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/theme/src/components/modal.ts).
```

Comment:
I've prepared an illustration for you. Put the `AspectRatio` below the anatomy list (see the Input or Menu theming docs to see exactly here). Here's the code for the illustration: https://gist.github.com/nikolovlazar/bb69fcd8d035bad0adbf372e033f86ff

Also add letters to the parts:
```
- A: `header`
- B: `overlay`
- C: `dialogContainer`
- D: `dialog`
- E: `closeButton`
- F: `body`
- G: `footer`
```